enclave/core/sgx/tracee.h

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_CORE_SGX_TRACEE_H
+#define _OE_CORE_SGX_TRACEE_H
+
+#include <openenclave/corelibc/stdbool.h>
+#include <openenclave/internal/sgx/td.h>
+
+bool is_enclave_debug_allowed_with_td(oe_sgx_td_t* td);
```

Comment:
Debugging is an enclave level concept, and it reads odd that such a query is tied to the notion of threads.